NFig/SettingsFactory.cs

```diff
@@ -1214,6 +1227,11 @@ PropertyToSettingDelegate GetPropertyToSettingDelegate(Type type)
 
         delegate TValue SettingGetterDelegate<TValue>(TSettings settings);
 
+        static FieldInfo GetGetterOnlyBackingField(PropertyInfo property)
+        {
+            return property.DeclaringType.GetField("<" + property.Name + ">k__BackingField", BindingFlags.Instance | BindingFlags.NonPublic);
```

Comment:
Oof.. I know this _works_, but relying on the compiler generated name of a backing field feels _super_ brittle to me. I have no clude if this would work in .NET Core (which we plan to support). Is this really the _only_ way to get a backing field for a get-only property?